homeassistant/components/scrape/__init__.py

```diff
@@ -70,17 +72,27 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
                 seconds=resource_config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
             ),
         )
-        await coordinator.async_refresh()
-        if coordinator.data is None:
-            raise PlatformNotReady
 
+        refresh_coroutines.append(coordinator.async_refresh())
+
+        sensors: list[ConfigType] = resource_config.get(SENSOR_DOMAIN, [])
         for sensor_config in sensors:
-            discovery.load_platform(
-                hass,
-                Platform.SENSOR,
-                DOMAIN,
-                {"coordinator": coordinator, "config": sensor_config},
-                config,
+            load_coroutines.append(
+                discovery.async_load_platform(
+                    hass,
+                    Platform.SENSOR,
+                    DOMAIN,
+                    {"coordinator": coordinator, "config": sensor_config},
+                    config,
+                )
             )
+    print(refresh_coroutines)
+    print(load_coroutines)
+    print(config)
```

Comment:
`print`?

Reply:
Fixing the tests. I will let you know when it's complete